flocker/control/_persistence.py

```diff
@@ -0,0 +1,86 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Persistence of cluster configuration.
+"""
+
+from cPickle import dumps, loads
+
+from twisted.application.service import Service
+
+from ._model import Deployment
+
+
+# These should not use Pickle!@!
+# https://clusterhq.atlassian.net/browse/FLOC-1241x
+def serialize_deployment(deployment):
+    """
+    Convert a ``Deployment`` object to ``bytes``.
+
+    :param Deployment deployment: Object to serialize.
+
+    :return bytes: Serialized object.
+    """
+    return dumps(deployment)
+
+
+def deserialize_deployment(data):
+    """
+    Create a ``Deployment`` object that was previously serialized to given
+    ``bytes``.
+
+    :param bytes data: Output of ``serialize_deployment``.
+
+    :return Deployment: Deserialized object.
+    """
+    return loads(data)
+
+
+class ConfigurationPersistenceService(Service):
+    """
+    Persist configuration to disk, and load it back.
+
+    :ivar Deployment _deployment: The current desired deployment configuration.
+    """
+    def __init__(self, reactor, path):
+        """
+        :param reactor: Reactor to use for thread pool.
```

Comment:
Oh yeah, I was going to file another issue, right. Unless you want this done in this branch.